### PR TITLE
Make databricks jobs reset match design.

### PR DESCRIPTION
**Pull request by andrewmchen**
`databricks jobs reset` should take two parameters --json and --job-id. Then it should materialize the request body sent to the reset endpoint with
```
{
  "job_id": job_id,
  "new_settings": json
}
```

### CODE REVIEW

File: databricks_cli/jobs/cli.py [hunk 1]
```diff
@@ -47,10 +47,25 @@ def create_cli(json_file, json):
...
-    json_cli_base(json_file, json, reset_job)
+def reset_cli(json_file, json, job_id):
+    if not bool(json_file) ^ bool(json):
+        raise RuntimeError('Either --json-file or --json should be provided')
+    if json_file:
+        with open(json_file, 'r') as f:
+            json = f.read()
+    deser_json = json_loads(json)
+    if 'job_id' in deser_json:
+        raise RuntimeError('job_id should not be provided in the json spec. The json spec should ' +
```
Comment (loneknightpy) [inline databricks_cli/jobs/cli.py]: This check is not needed.

Comment (andrewmchen) [reply databricks_cli/jobs/cli.py]: It's a common error to put the job_id in the --json option so I thought it'd be nice to catch it in the client side.

Comment (loneknightpy) [reply databricks_cli/jobs/cli.py]: why is it invalid? If you send the request to server, it will accept it.

**Comment by loneknightpy** [comment]
lgtm

**Comment by fmms** [comment]
@andrewmchen , @loneknightpy this commit broke the documentation. Can you comment in #356 how to get that updated. Thanks.

**Pull request merged**